pkgs/test_core/lib/src/runner/reporter/json.dart

```diff
@@ -297,8 +303,21 @@ class JsonReporter implements Reporter {
   /// If javascript traces are enabled and the test is on a javascript platform,
   /// or if the [trace] is null or empty, then the line, column, and url will
   /// all be `null`.
-  Map<String, dynamic> _frameInfo(SuiteConfiguration suiteConfig, Trace? trace,
-      SuitePlatform platform, String suitePath) {
+  Map<String, dynamic> _locationInfo(
+      SuiteConfiguration suiteConfig,
+      Trace? trace,
+      TestLocation? location,
+      SuitePlatform platform,
+      String suitePath) {
+    // If this test has a location override, always use that.
+    if (location != null) {
+      return {
```

Comment:
can this just be `location.serialize`?

Reply:
We could, but it would be easy to accidentally change the JSON reporter output when changing the internal classes - I don't know if that's reasonable or not. If you think it is, I'll change this and also include a note in the `TestLocation` `serialize`/`deserialize` methods about that.

Reply:
I think I would just add the comment to the TestLocation methods and call `serialize` here - otherwise if we add fields or something to TestLocation, we might forget to update this, so there is some risk either way.

Reply:
Done!